presentation/src/main/java/com/d83t/bpm/presentation/ui/splash/SplashViewModel.kt

```diff
@@ -0,0 +1,40 @@
+package com.d83t.bpm.presentation.ui.splash
+
+import androidx.lifecycle.viewModelScope
+import com.d83t.bpm.domain.usecase.GetStoredIdUseCase
+import com.d83t.bpm.presentation.base.BaseViewModel
+import com.d83t.bpm.presentation.di.IoDispatcher
+import com.d83t.bpm.presentation.di.MainDispatcher
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import javax.inject.Inject
+
+@HiltViewModel
+class SplashViewModel @Inject constructor(
+    private val getStoredIdUseCase: GetStoredIdUseCase,
+    @MainDispatcher private val mainDispatcher: CoroutineDispatcher,
+    @IoDispatcher private val ioDispatcher: CoroutineDispatcher
+) : BaseViewModel() {
+
+    private val _event = MutableSharedFlow<SplashViewEvent>()
+    val event: SharedFlow<SplashViewEvent>
+        get() = _event
+
+    private val _state = MutableStateFlow<SplashState>(SplashState.Init)
+    val state: StateFlow<SplashState>
+        get() = _state
+
+    fun getStoredId() {
+        val storedIdFlow = getStoredIdUseCase()
+
+        viewModelScope.launch(ioDispatcher) {
+            val kakaoId = storedIdFlow.first()
```

Comment:
이 부분 별도의 변수로 처리 하지 않고 이렇게도 활용 가능할 듯 합니다 :) invoke 형식으로 유즈케이스를 만들었으니 ㅎㅎ
```suggestion
            val kakaoId = getStoredIdUseCase().first()
```